src/site/_drafts/perception-toolkit/index.md

```diff
@@ -0,0 +1,161 @@
+you to it.
+
+## How does it work?
+
+The open-source Web Perception Toolkit helps you add visual search to your
+website. It passes a device camera stream through a set of detectors that map
+markers or targets to structured data on your site. With this data, you can
+present the right information in a customizable UI.
+
+I'll show you enough of this to give you a taste of how it works. For a complete
+explanation, check out the Getting Started Guide, the toolkit reference, and the
+demos.
+
+## Customize the user experience
+
+Getting UI elements is amazingly simple. The toolkit gives you livecycle events,
+Card and Button objects for crafting the user experience around those events,
+and an easy way to style the cards.
+
+The most important lifecycle event is MarkerChanges, which is called every time
+a marker is found. A marker is the toolkit's term for one of the real-world
+objects you're using the toolkit to look for. A marker can be a physical object,
+an image, a barcode, or a QR code.
+
+The process for responding to the event is the same as for any other event.
+
+```js
+const container = document.querySelector('.container');
+async function onMarkerChanges(event) {
+  // Process the event.
+}
+window.addEventListener(PerceptionToolkit.Events.MarkerChanges, onMarkerChanges);
+```
+
+Let's look at the event more closely. The event itself contains arrays of
+markers it's both found and lost. This helps account for hand and marker
+movements: cameras not help steadily enough, dropped markers, that kind of
+thing. As long as the toolkit finds the marker at some point, it will be in
+either array.
+
+```js
+async function onMarkerChanges(event) {
+  event.preventDefault();
+  if (container. childNodes.length > 0) { return; }
+  const { found, lost } = event.detail;
+  // Deal with lost and found objects.
+}
+```
+
+Next, we show an appropriate card based on what the toolkit found. Notice the
+conveniences I've added to the user experience. Whether the marker is found or
+not, I provide one-click access to what I think is most useful in the
+circumstances.
+
+```js
+async function onMarkerChanges(event) {
+  event.preventDefault();
+  if (container. childNodes.length > 0) { return; }
+  const { found, lost } = event.detail;
+  const { Card } = PerceptionToolkit.Elements;
+  if (found.length === 0 && lost.length === 0) {
+    const card = new Card();
+    card.src = 'We wish we could help, but that\'s not our razor. Would you like to see <a href="/catalog">our catalog</a>?';
+    card.dataset.notRecognized = true;
+    container.appendChild(card);
+  } else if (found.length > 0 || lost.length > 0) {
+    const card = new Card();
+    card.src = found[0].content || lost[0].content;
+    // The ActionButton object is also part of the toolkit.
+    // Its instantiation is not shown.
+    carc.appendChild(reorderActionButton);
+    container.appendChild(card);
+  }
+}
+```
```

Comment:
The end of this snippet seems like a sweet spot for an example image or gif of the experience running.